src/server/server.cc

```diff
@@ -1753,3 +1755,52 @@ std::list<std::pair<std::string, uint32_t>> Server::GetSlaveHostAndPort() {
   slave_threads_mu_.unlock();
   return result;
 }
+
+// The numeric cursor consists of a 32-bit hash, a 16-bit time stamp, and a 16-bit counter, with the highest bit set to
+// 1 to prevent a zero cursor from occurring. The hash is used to prevent information leakage. The time_stamp is used to
+// prevent the generation of the same cursor in the extremely short period before and after a restart.
+static uint64_t GetNumberCursor(const std::string &key_name, uint16_t counter) {
+  auto hash = static_cast<uint32_t>(std::hash<std::string>{}(key_name));
+  auto time_stamp = static_cast<uint16_t>(util::GetTimeStamp());
+  // set first bit 1, so number cursor not be 0
+  return static_cast<uint64_t>(hash) | static_cast<uint64_t>(time_stamp) << 32 | static_cast<uint64_t>(counter) << 48 |
+         static_cast<uint64_t>(1) << 63;
```

Comment:
And can we don't set the first bit to 1? If the counter is zero, we skip this?

Reply:
I don't like skip zero counter, as it will make the code more complicated.
Maybe we can set the first bit of the `hash` to 1 instead of `counter`.

Reply:
👌I get it. Just asking.

Reply:
I just wondering you use GetIndexFromNumberCursor() to generate the index of the array `cursor_dict_`. It seems we will ignore the high bit of the index because we module the num by `CURSOR_DICT_SIZE`. So the key point is that we need the `1` to block the `numeric cursor` from 0 and we will reduce the confilct if we put the bit to the counter.

Reply:
That's right. However, in reality, the difference is not very big, and each has its own advantages and disadvantages. 
For me, it makes no difference where `1` is placed.

Reply:
I get it.

Reply:
Using counter is zero to marking means that when it overflow, the number will back to 1, I guess when ring size is 2**14, there are only slightly differences...